### DISCUSSION TITLE
Too many bugs

### DISCUSSION THREAD

**Discussion by MingyuShao** [thu-ml/tianshou#1037] [Q&A] [answered]
When I first download tianshou, I ran the test case in test/DQN and it failed:
![0b549f73b8d10c4c4e11d5aebbd91ae5](https://github.com/thu-ml/tianshou/assets/156297397/e707909f-fb2d-4aec-a70c-7561c4c490d3)
and when I tried to run other cases in the test folder, they all failed with bugs:
![f5f50889d6522f789dcd46755214668f](https://github.com/thu-ml/tianshou/assets/156297397/f728a571-2954-4460-b71d-4790be83f25a)
![image](https://github.com/thu-ml/tianshou/assets/156297397/a6c5e55a-1eac-4cbb-9934-ac7569b9c123)
Is some error in my installation? I'm looking forward to help, thanks.

**Comment by opcode81** [comment, marked as answer]
Could it be that you have an (old) installation of tianshou in your Python environment which is being accessed instead of the source version?
Did you install the source version of tianshou via poetry?

Also, please prefer copies of the text content from your console to screenshots. The screenshots are incomplete and therefore the context information is insufficient.
